### ISSUE TITLE
E2E: Evaluate Ingestion of End to End Test Data Into BigQuery

### ISSUE THREAD

**Issue by pjhill** [department-of-veterans-affairs/va.gov-team#31992]
## User Story

_E2E test data is currently being generated using Mochawesome, and a data set of those results is being pushed to BigQuery. This has served our current, MVP dashboaridng needs. As we expand and build this out to a larger dashboard, we should reevaluate that the shape of the data we are pushing to BigQuery will serve all the metrics we are looking to track. We may need to consider breaking apart the E2E data into two separately shaped datasets, depending on the kind of access we need to the data for easier management. It is important to make these determinations ahead of time as we cannot change a BQ schema once it's loaded with data without wiping all the historical data._

## Tasks
- [x] Evaluate current E2E test data ingestion mechanism to see if changes need to be made to account for metrics we cannot currently track.
- [x] Data should be ingested for all applications in one data set, so a Platform-level overview can be given. There should be a team and application associated with each row, so the charts can be properly filtered down to a single application/team view in Domo.
- [x] Adjust BigQuery Schemas and/or create new BigQuery tables as needed to meet the needs of our new metrics.
- [x] Evaluate our current CI pipeline and make any adjustments necessary to ensure that the data is sent to BigQuery every CI run.
- [ ] Create Domo data connections to BQ that pull these data sets to Domo to be ready for cards.

## Acceptance Criteria

- [ ] _Ingestion mechanism provides all the data we need to produce our desired E2E metrics in usable formats._
- [ ] _CI workflow updates this data every run._
- [ ] _Domo has data connections for all the E2E test data that can be used by our cards._

**Comment by CBonade** [comment]
Some preliminary queries today and base scaffolding of a couple Domo cards on the product level. Getting to a point of learning how to do the drill-downs and figuring out good ways to display this data. While the cards ARE a separate ticket initiative, without at least a couple passes at what the cards need to look like it wouldn't be possible to guarantee we have all the data necessary. Once one of our more complex cards can be constructed with the data we have, I'll mark this as ready to go.

**Comment by CBonade** [comment]
All metrics are currently being tracked with the exception of test flakiness. After many different attempted approaches I believe the approach of tracking retries in a separate BigQuery table is going to serve us well. I currently have retry data making it to our dashboard data repository and am tinkering with the formatting to get it to make it up to BigQuery accordingly.

**Comment by CBonade** [comment]
Today the data ingestion process to BigQuery for the flakiness data was completed. Unfortunately, this presented a new complication with regards to the Cypress Test Results data. The complication surrounded the existing handling of the context object that was being passed via Mochawesome's `addContext` function. As we prior were only using this for video URLs of failed videos, we now have to use that for all retry data as well. The video link was set up to completely overwrite the context object, so I took the approach of instead of just uploading the retry data from the context object directly, to split those properties out and add them to the test result. However, this presented the problem of conflicting with the Cypress Test Result table schema, so I've added in a followup function that removes these properties once they've been sent to BQ. There was an additional bug to fix surrounding contract tests but this is in it's final testing stages before it's moved to review. Just need to test a purposeful failure in a test, and also implement an intentional retry as well. Going to hold til the morning to merge anything, as it requires a coordinated merge of `vets-website` and `testing-tools-team-data-dashboard` repo PRs at a similar time and monitoring to ensure nothing goes wrong.
